pkg/appStore/deployment/repository/InstalledAppRepository.go

```diff
@@ -545,3 +559,100 @@ func (impl InstalledAppRepositoryImpl) GetInstalledAppByGitHash(gitHash string)
 	}
 	return model, nil
 }
+
+func (impl InstalledAppRepositoryImpl) GetInstalledAppByAppId(appId int) (InstalledApps, error) {
+	var installedApps InstalledApps
+	queryString := `select * from installed_apps where active=? and app_id=? and deployment_app_type=?;`
+	_, err := impl.dbConnection.Query(&installedApps, queryString, true, appId, util2.PIPELINE_DEPLOYMENT_TYPE_ACD)
+	if err != nil {
+		impl.Logger.Errorw("error in fetching InstalledApp", "err", err)
+		return installedApps, err
+	}
+
+	return installedApps, nil
+}
+
+func (impl InstalledAppRepositoryImpl) GetInstalledAppByInstalledAppVersionId(installedAppVersionId int) (InstalledApps, error) {
+	var installedApps InstalledApps
+	queryString := `select ia.* from installed_apps ia inner join installed_app_versions iav on ia.id=iav.installed_app_id
+         			where iav.active=? and iav.id=? and ia.deployment_app_type=?;`
+	_, err := impl.dbConnection.Query(&installedApps, queryString, true, installedAppVersionId, util2.PIPELINE_DEPLOYMENT_TYPE_ACD)
+	if err != nil {
+		impl.Logger.Errorw("error in fetching InstalledApp", "err", err)
+		return installedApps, err
+	}
+
+	return installedApps, nil
+}
+
+func (impl InstalledAppRepositoryImpl) GetAllGitOpsDeploymentAppName() ([]string, error) {
+	type GitOpsAppName struct {
+		GitOpsAppName string `sql:"git_ops_app_name"`
+	}
+	var gitOpsApplicationName []*GitOpsAppName
+	allGitOpsAppName := make([]string, 0)
+
+	query := `select concat(a.git_ops_repo_name, '-',e.environment_name) as git_ops_app_name from installed_apps a inner join environment e on a.environment_id=e.id;`
+	_, err := impl.dbConnection.Query(&gitOpsApplicationName, query)
+	if err != nil {
+		impl.Logger.Errorw("error in GetAllGitOpsDeploymentAppName", "err", err)
+		return nil, err
+	}
+
+	for _, item := range gitOpsApplicationName {
+		allGitOpsAppName = append(allGitOpsAppName, item.GitOpsAppName)
+	}
+	return allGitOpsAppName, err
+}
+
+func (impl InstalledAppRepositoryImpl) GetAllGitOpsAppNameAndInstalledAppMapping() ([]*GitOpsAppDetails, error) {
+	var model []*GitOpsAppDetails
+
+	query := `select concat(a.git_ops_repo_name, '-',e.environment_name) as git_ops_app_name, a.id as installed_app_id from installed_apps a 
+    			inner join environment e on a.environment_id=e.id where a.active=true and e.active=true;`
+	_, err := impl.dbConnection.Query(&model, query)
+	if err != nil {
+		impl.Logger.Errorw("error in GetAllGitOpsDeploymentAppName", "err", err)
+		return nil, err
+	}
+	return model, err
+}
+
+func (impl InstalledAppRepositoryImpl) GetArgoPipelinesHavingLatestTriggerStuckInNonTerminalStatusesForAppStore(getPipelineDeployedBeforeMinutes int, getPipelineDeployedWithinHours int) ([]*InstalledAppVersions, error) {
+	var installedAppVersions []*InstalledAppVersions
+	queryString := `select iav.* from installed_app_versions iav 
```

Comment:
avoid native queries

Reply:
noted, although for complex sub queries I tend to use native queries since they become complex and is easily deciphered by seeing the query.

Reply:
have used pg client at places where I could have